### PR TITLE
Fix XM Full netstandard to copy in correct assemblies

### PR DESCRIPTION
**Pull request by chamons**
- Applications will build but crash on launch otherwise

### CODE REVIEW

**Comment by chamons** [comment]
I caught this while writing an auto test.

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets [hunk 1]
```diff
@@ -536,6 +536,9 @@ Copyright (C) 2014 Xamarin. All rights reserved.
 	<Target Name="_CompileToNative" DependsOnTargets="_DetectAppManifest;_DetectSdkLocations;_GenerateBundleName;ResolveReferences;_CompileEntitlements;_CompileAppManifest"
 		Inputs="$(TargetDir)$(TargetFileName)"
 		Outputs="$(_AppBundlePath)Contents\MacOS\$(_AppBundleName)">
+		<ItemGroup>
+			<ReferencesOnly Include="@(ReferenceCopyLocalPaths)" Condition="'%(ReferenceCopyLocalPaths.Extension)' == '.dll'" />
```
Comment (radical) [inline msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: Maybe name like `ReferenceCopyLocalAssemblyPaths`?

Comment (jstedfast) [reply msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: Wouldn't that just append to the existing item group of that name?

Comment (jstedfast) [reply msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: oh, woops, the existing name doesn't have the word Assembly in it. n/m :)

**Comment by radical** [comment]
and a commit message like:

```
[msbuild] Fix XM Full netstandard to copy in correct assemblies

nuget packages can have `ref` (reference assemblies) and `lib` (implementation assemblies),
with the former meant for use with the compiler and the latter required for execution.

The `ResolveAssemblyReference` task populates `@(ReferencePath)` with the assemblies meant
for use with the compiler, so these get the `ref` assemblies. And it populates `@(ReferenceCopyLocalPaths)`
with any assemblies or files related (.pdb/.mdb/.xml etc) to the references, and this one gets the `lib`
assemblies.

XM needs to bundle the assemblies required for execution, but it uses `@(ReferencePath)` and thus
ends up with reference assembly, which would fail to load at runtime! Instead, we extract the `.dll`
files from `@(ReferenceCopyLocalPaths)` and use that. 
```

File: msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets (hunk 1 above)
Comment (radical) [reply msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets]: yeah, I just wanted to make it easy to see where that extracted item was sourced from!

**Comment by chamons** [comment]
I'll steal that when i squash merge.

**Comment by chamons** [comment]
build

**Comment by chamons** [comment]
Looks like the breakages are non-trivial, will have to dig into them.

**Comment by chamons** [comment]
Ok, that fixes all tests locally _and_ them when I hack it to run msbuild (https://gist.github.com/chamons/8fb988deb4daca052663efa116cea084).

I think the change is safe to stew in master, but i'm questioning 15.4...

**Comment by chamons** [comment]
This is strange, I can build things just fine locally, even with no XM installed. Let's try one more time to verify before I start digging up the world...

**Comment by chamons** [comment]
build

**Comment by chamons** [comment]
Ok, somehow this broke build machine builds but not my local builds.

Digging...

**Comment by chamons** [comment]
I can reproduce if I git clean external/guiunit

**Comment by chamons** [comment]
Test failure is known - https://bugzilla.xamarin.com/show_bug.cgi?id=59277

**Comment by chamons** [comment]
I feel more confident landing this, as it is nearly identical to what XI does.

**Pull request merged**